### Nombre: Carla Morales López
### Matrícula: A01639225

# Undertanding the cartwheel data set

The notebook aims to undertand the content of the cartwheel data set.


## Acknowledgments

- Data from https://www.coursera.org/ from the course "Understanding and Visualizing Data with Python" by University of Michigan


# Cartwheel data set

1. A cartwheel

![cartwheel1.png](datasets/cartwheel/cartwheel1.png)

2. The dataset description
    - The dataset used here is an extension from the original cartwheel dataset from cursera
    - Total number of observations: 28
    - Many observations/measurements/recordings of the characteristics/attributes/variables of cartwheel executions
    - Variables: Age, Gender, GenderGroup, Glasses, GlassesGroup, Height, Wingspan, CWDistance, ... (X variables)
    


# Importing and inspecting the data

In [103]:
# Import the packages that we will be using

import pandas as pd

# Dataset url

url_cartwheel = "datasets/cartwheel/cartwheel.csv"

# Load the dataset

dataset_cartwheel = pd.read_csv(url_cartwheel)

dataset_cartwheel

ID   Age Gender  GenderGroup Glasses  GlassesGroup  Height  Wingspan  \
0    1  56.0      F            1       Y             1   62.00      61.0   
1    2  26.0      F            1       Y             1   62.00      60.0   
2    3  33.0      F            1       Y             1   66.00      64.0   
3    4  39.0      F            1       N             0   64.00      63.0   
4    5  27.0      M            2       N             0   73.00      75.0   
5    6  24.0      M            2       N             0   75.00      71.0   
6    7  28.0      M            2       N             0   75.00      76.0   
7    8  22.0      F            1       N             0   65.00      62.0   
8    9  29.0      M            2       Y             1   74.00      73.0   
9   10  33.0      F            1       Y             1   63.00      60.0   
10  11  30.0      M            2       Y             1   69.50      66.0   
11  12  28.0      F            1       Y             1   62.75      58.0   
12  13  25.0      F            1       Y             1   65.00      64.5   
13  14  23.0      F            1       N             0   61.50      57.5   
14  15  31.0      M            2       Y             1   73.00      74.0   
15  16  26.0      M            2       Y             1   71.00      72.0   
16  17  26.0      F            1       N             0   61.50      59.5   
17  18  27.0      M            2       N             0   66.00      66.0   
18  19  23.0      M            2       Y             1   70.00      69.0   
19  20  24.0      F            1       Y             1   68.00      66.0   
20  21  23.0      M            2       Y             1   69.00      67.0   
21  22  29.0      M            2       N             0   71.00      70.0   
22  23  25.0      M            2       N             0   70.00      68.0   
23  24  26.0      M            2       N             0   69.00      71.0   
24  25  23.0      F            1       Y             1   65.00      63.0   
25  26  30.0      F            1       N             0   74.60       NaN   
26  27   NaN      M            2       N             0   71.00      70.0   
27  28  27.0      M            2       N             0     NaN      71.5   

    CWDistance Complete  CompleteGroup  Score  
0           79        Y            1.0      7  
1           70        Y            1.0      8  
2           85        Y            1.0      7  
3           87        Y            1.0     10  
4           72        N            0.0      4  
5           81        N            0.0      3  
6          107        Y            1.0     10  
7           98        Y            1.0      9  
8          106        N            0.0      5  
9           65        Y            1.0      8  
10          96        Y            1.0      6  
11          79        Y            1.0     10  
12          92        Y            1.0      6  
13          66        Y            1.0      4  
14          72        Y            1.0      9  
15         115        Y            1.0      6  
16          90        N            0.0     10  
17          74        Y            1.0      5  
18          64        Y            1.0      3  
19          85        Y            1.0      8  
20          66        N            0.0      2  
21         101        Y            1.0      8  
22          82        Y            1.0      4  
23          63        Y            1.0      5  
24          67        N            0.0      3  
25          71        Y            1.0      9  
26         101        Y            NaN      8  
27         103        Y            1.0     10

# Final remarks


- The **Pandas** library provides fancy, high-performance, easy-to-use data structures and data analysis tools


# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and compare with the data printed above.

In [104]:
url_iris = "datasets/iris/iris.csv"
dataset_iris = pd.read_csv(url_iris, header=None,
                           names=["Sepal Length", "Sepal Width", "Petal Length", "Petal Width", "Class"])
dataset_iris

Sepal Length  Sepal Width  Petal Length  Petal Width           Class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

2. How many observations (rows) are in total?

In [105]:
len(dataset_iris.index)

150

***Total Number of Rows = 150***

3. How many variables (columns) are in total? What do they represent?

In [106]:
len(dataset_iris.columns)  # Columns = 5

5

***Columns***
 1. Sepal Length in cm
 2. Sepal Width in cm
 3. Petal Length in cm
 4. Petal Width in cm
 5. Class:
     * Iris Setosa
     * Iris Versicolour
     * Iris Virginica

4. How many observations are for each type of flower?

In [107]:
dataset_iris.groupby(['Class']).size()

Class
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
dtype: int64

5. What is the type of data for each variable?

In [108]:
dataset_iris.dtypes

Sepal Length    float64
Sepal Width     float64
Petal Length    float64
Petal Width     float64
Class            object
dtype: object

6. What are the units of each variable?

    1. CM
    2. CM
    3. CM
    4. CM
    5. {"Iris-setosa","Iris-virginica","Iris-versicolor"}